metadata-ingestion/src/datahub/ingestion/source/sql/vertica.py

```diff
@@ -18,254 +28,1381 @@
+
+                if sql_config.include_views:
+                    yield from self.loop_views(inspector, schema, sql_config)
+
+                if sql_config.include_projections:
+                    yield from self.loop_projections(inspector, schema, sql_config)
+                if sql_config.include_models:
+                    yield from self.loop_models(inspector, schema, sql_config)
+
+                if profiler:
+                    profile_requests += list(
+                        self.loop_profiler_requests(inspector, schema, sql_config)
+                    )
+                    profile_requests += list(
+                        self.loop_profiler_requests_for_projections(
+                            inspector, schema, sql_config
+                        )
+                    )
+
+            if profiler and profile_requests:
+                yield from self.loop_profiler(
+                    profile_requests, profiler, platform=self.platform
+                )
+
+            oauth_schema = "Entities"
+            if sql_config.include_oauth:
+                yield from self.loop_Oauth(inspector, oauth_schema, sql_config)
+
+        # Clean up stale entities.
+        yield from self.stale_entity_removal_handler.gen_removed_entity_workunits()
+
+    def get_database_properties(
+        self, inspector: Inspector, database: str
+    ) -> Optional[Dict[str, str]]:
+        try:
+            custom_properties = inspector._get_database_properties(database)
+            return custom_properties
+        except Exception as ex:
+            self.report.report_failure(
+                f"{database}", f"unable to get extra_properties : {ex}"
+            )
+
+    def get_schema_properties(
+        self, inspector: Inspector, database: str, schema: str
+    ) -> Optional[Dict[str, str]]:
+        try:
+            custom_properties = inspector._get_schema_properties(schema)
+            return custom_properties
+        except Exception as ex:
+            self.report.report_failure(
+                f"{schema}", f"unable to get extra_properties : {ex}"
+            )
+
+    def _process_table(
+        self,
+        dataset_name: str,
+        inspector: Inspector,
+        schema: str,
+        table: str,
+        sql_config: SQLAlchemyConfig,
+    ) -> Iterable[Union[SqlWorkUnit, MetadataWorkUnit]]:
+        dataset_urn = make_dataset_urn_with_platform_instance(
+            self.platform,
+            dataset_name,
+            self.config.platform_instance,
+            self.config.env,
+        )
+        table_owner = self._get_owner_information(table, "table")
+        yield from add_owner_to_entity_wu(
+            entity_type="dataset",
+            entity_urn=dataset_urn,
+            owner_urn=f"urn:li:corpuser:{table_owner}",
+        )
+        yield from super()._process_table(
+            dataset_name, inspector, schema, table, sql_config
+        )
+
+    def loop_views(
```

Comment:
what was changed in this method?

it should be possible to simplify similar to how loop_tables was simplified

Reply:
if sql_config.include_view_lineage: added which checks and ingests lineage for views . since this is inside forloop we were not able to use super